FixDevV3Blobs/FixDevV3Blobs/Program.cs

```
@@ -0,0 +1,192 @@
+            stopwatch.Stop();
+            return blobList;
+        }
+
+        private static async Task<Stopwatch> ProcessBlobList(List<CloudBlockBlob> blobList, string search, string replace)
+        {
+            var stopwatch = Stopwatch.StartNew();
+            var tasks = Enumerable.Range(1, MaxTasks).Select(_ => Task.CompletedTask).ToArray();
+            var count = 0;
+            var skipped = 0;
+            foreach (var blob in blobList)
+            {
+                ++count;
+                //await ProcessBlob(blob);
+                if (Processed(blob))
+                {
+                    ++skipped;
+                    continue;
+                }
+                await StartProcessing(tasks, blob, search, replace);
+                if ((count % 1000) == 0)
+                {
+                    Log(
+                        "{0}/{1} (skipped: {4}) {2}, ETA: {3}",
+                        count,
+                        blobList.Count,
+                        stopwatch.Elapsed,
+                        TimeSpan.FromSeconds(((double)blobList.Count - count) / count * stopwatch.Elapsed.TotalSeconds),
+                        skipped);
+                }
+            }
+            await Task.WhenAll(tasks);
+            stopwatch.Stop();
+            return stopwatch;
+        }
+
+        private static bool Processed(CloudBlockBlob blob)
+        {
+            return ProcessedBlobs.Contains(blob.Uri.AbsoluteUri);
+        }
+
+        private static async Task StartProcessing(Task[] tasks, CloudBlockBlob blob, string search, string replace)
+        {
+            await Task.WhenAny(tasks);
+
+            for (int i = 0; i < tasks.Length; ++i)
+            {
+                if (tasks[i].IsCompleted)
+                {
+                    tasks[i] = ProcessBlob(blob, search, replace);
+                    return;
+                }
+            }
+
+            throw new Exception("Failed to find completed task");
```

nit: please throw `InvalidOperationException`. using base exception is a bad practice since it makes the catch site very unclear and overly broad